tiny_dnn/core/kernels/conv2d_op_avx.h

```diff
@@ -313,55 +324,59 @@ void avx_conv2d_5x5_kernel(const core::conv_params &params,
-        __m128d i6       = _mm_load_sd(pi + 24);
-        __m256d tmp0     = _mm256_mul_pd(w0, i0);
-        __m256d tmp1     = _mm256_mul_pd(w1, i1);
-        __m256d tmp2     = _mm256_mul_pd(w2, i2);
-        __m256d tmp3     = _mm256_mul_pd(w3, i3);
-        __m256d tmp4     = _mm256_mul_pd(w4, i4);
-        __m256d tmp5     = _mm256_mul_pd(w5, i5);
-        __m128d tmp6     = _mm_mul_pd(w6, i6);
-        sum0             = _mm256_add_pd(tmp0, sum0);
-        sum1             = _mm256_add_pd(tmp1, sum1);
-        sum2             = _mm256_add_pd(tmp2, sum2);
-        sum3             = _mm256_add_pd(tmp3, sum3);
-        sum4             = _mm256_add_pd(tmp4, sum4);
-        sum5             = _mm256_add_pd(tmp5, sum5);
-        sum6             = _mm_add_pd(tmp6, sum6);
+        __m256d w0 = _mm256_loadu_pd(pw + 0);
+        __m256d w1 = _mm256_loadu_pd(pw + 4);
+        __m256d w2 = _mm256_loadu_pd(pw + 8);
+        __m256d w3 = _mm256_loadu_pd(pw + 12);
+        __m256d w4 = _mm256_loadu_pd(pw + 16);
+        __m256d w5 = _mm256_loadu_pd(pw + 20);
+        __m128d w6 = _mm_load_sd(pw + 24);
+
+        const double *pi = &i_begin[inidx];
+
+        __m256d i0   = _mm256_loadu_pd(pi + 0);
+        __m256d i1   = _mm256_loadu_pd(pi + 4);
+        __m256d i2   = _mm256_loadu_pd(pi + 8);
+        __m256d i3   = _mm256_loadu_pd(pi + 12);
+        __m256d i4   = _mm256_loadu_pd(pi + 16);
+        __m256d i5   = _mm256_loadu_pd(pi + 20);
+        __m128d i6   = _mm_load_sd(pi + 24);
+        __m256d tmp0 = _mm256_mul_pd(w0, i0);
+        __m256d tmp1 = _mm256_mul_pd(w1, i1);
+        __m256d tmp2 = _mm256_mul_pd(w2, i2);
+        __m256d tmp3 = _mm256_mul_pd(w3, i3);
+        __m256d tmp4 = _mm256_mul_pd(w4, i4);
+        __m256d tmp5 = _mm256_mul_pd(w5, i5);
+        __m128d tmp6 = _mm_mul_pd(w6, i6);
+        sum0         = _mm256_add_pd(tmp0, sum0);
+        sum1         = _mm256_add_pd(tmp1, sum1);
+        sum2         = _mm256_add_pd(tmp2, sum2);
+        sum3         = _mm256_add_pd(tmp3, sum3);
+        sum4         = _mm256_add_pd(tmp4, sum4);
+        sum5         = _mm256_add_pd(tmp5, sum5);
+        sum6         = _mm_add_pd(tmp6, sum6);
       }
       sum0         = _mm256_add_pd(sum0, sum1);
       sum2         = _mm256_add_pd(sum2, sum3);
       sum4         = _mm256_add_pd(sum4, sum5);
       sum0         = _mm256_add_pd(sum0, sum2);
       __m256d sum  = _mm256_add_pd(sum0, sum4);
-      __m128d b    = _mm_load_sd(&bias[o]);
+      __m128d b    = _mm_load_sd(&bias_begin[o]);
       __m128d hsum = hsum256_pd(sum);
       b            = madd128_sd(b, y_bias_scale, sum6);
-      _mm_store_sd(&a[o], _mm_add_sd(hsum, b));
+      _mm_store_sd(&a_begin[o], _mm_add_sd(hsum, b));
     }
   } else {
     for (serial_size_t o = 0; o < out.depth_; ++o, oidx += out_area) {
-      double *pa = &a[oidx];
-      double b   = bias[o] * bias_scale;
+      double *pa = &a_begin[oidx];
+      double b   = bias.host_at(0, o) * bias_scale;
```

Comment:
`host_at` method inside loop isn't good for performance and it can be easily avoided in this case.